src/main/java/com/thealgorithms/datastructures/graphs/WelshPowell.java

```diff
@@ -0,0 +1,73 @@
+package com.thealgorithms.datastructures.graphs;
+
+import java.util.LinkedList;
+
+/*
+ *  The Welsh-Powell algorithm is a graph coloring algorithm
+ *  used for coloring a graph with the minimum number of colors.
+ *  https://en.wikipedia.org/wiki/Graph_coloring
+ */
+public class WelshPowell {
+
+    public static class WPGraph {
+
+        private int numVer; // Number of vertices in the graph
+        public LinkedList<Integer>[] adjLists; // Adjacency lists for the graph
```

Comment:
First of all, this field should be `private`.

Why did you decided to use `LinkedList<Integer>[]`? Why not `HashSet<Integer>[]` or even `HashMap<Integer, HashSet<Integer>>`. When you will go for the first one, you will still have to deal with graphs of fixed number of nodes (like in current solution).

Reply:
to be honest i just had a better knoweledge of the Linkedlist and i think the way it works its quite simple. I am gonna to change my code though and use a hashset instead to follow your idea.

Reply:
Decision which data structure to use is usually crucial. Please notice that when you construct you graph, you use the [`contains` method of `LinkedList`](https://github.com/TheAlgorithms/Java/blob/a7f3fcdcf3669c4205ec65f5626b7b885993e289/src/main/java/com/thealgorithms/datastructures/graphs/WelshPowell.java#L30) which is `O(n)`. On the other hand, in the in the _main part_ of the algorithm, you just iterate though the `LinkedList`, which is fine.

For me, storing the adjacency list of a single node in a `HashSet` seems to be natural (no duplicates, constant access time, easy iteration).

Please let me know when you are ready for next _review iteration_ or when you will have some questions.

Reply:
i changed both codes to fit better to your propositions, when you have time have a look thanks.